ldms/src/sampler/zfs_leafvdevs/zfs_leafvdevs.c

```diff
@@ -0,0 +1,525 @@
+ ****************************************************************************/
+static int resize_metric_set()
+{
+        size_t previous_heap_size;
+        size_t new_heap_size;
+        int    rc = 0;
+
+        previous_heap_size = ldms_set_heap_size_get(sampler_base->set);
+        base_set_delete(sampler_base);
+
+        new_heap_size  = previous_heap_size;
+        new_heap_size += zpool_vdev_heap_sz;
+
+        if (base_set_new_heap(sampler_base, new_heap_size) == NULL) {
+            rc = errno;
+           ldmsd_log(LDMSD_LERROR,
+                     SAMP" : Failed to resize metric set heap: %d\n", errno);
+        } else {
+		log_fn(LDMSD_LDEBUG, "ldms resize of list successful\n");
+	}
+        return rc;
+}
+
+static int sample(struct ldmsd_sampler *self)
+{
+        int rc = 0;
+
+        base_sample_begin(sampler_base);
+
+	list_handle = ldms_metric_get(sampler_base->set, index_store.vdev_list_idx);
+        ldms_list_purge(sampler_base->set, list_handle);
+
+        rc = zpool_iter(g_zfs, get_leaf_stats, NULL);
+        if (rc != 0) {
+            log_fn(LDMSD_LERROR, SAMP" sample():zfs_pool print_stat() failed: %d\n", rc);
+            base_sample_end(sampler_base);
+            goto err1;
+        }
+	/* this is where the rubber meets the pavement */
+
+        base_sample_end(sampler_base);
+
+err1:
+        return rc;
+}
+
+static void term(struct ldmsd_plugin *self)
+{
+        log_fn(LDMSD_LDEBUG, SAMP" term() called\n");
+        base_set_delete(sampler_base);
+        base_del(sampler_base);
+        sampler_base = NULL;
+}
+
+static ldms_set_t get_set(struct ldmsd_sampler *self)
+{
+	return NULL;
+}
+
+static const char *usage(struct ldmsd_plugin *self)
+{
+        log_fn(LDMSD_LDEBUG, SAMP" usage() called\n");
+	return  "config name=" SAMP " " BASE_CONFIG_SYNOPSIS
+                BASE_CONFIG_DESC
+                ;
+}
+
+struct ldmsd_plugin *get_plugin(ldmsd_msg_log_f pf)
+{
+        static struct ldmsd_sampler plugin = {
+                .base = {
+                        .name = SAMP,
+                        .type = LDMSD_PLUGIN_SAMPLER,
+                        .term = term,
+                        .config = config,
+                        .usage = usage,
+                },
+                .get_set = get_set,
+                .sample = sample,
+        };
+
+        log_fn = pf;
+        log_fn(LDMSD_LDEBUG, SAMP" get_plugin() called ("PACKAGE_STRING")\n");
+
+        return &plugin.base;
+}
+
+
+/*
+ * get a string suitable for an influxdb tag that describes this vdev
+ *
+ * By default only the vdev hierarchical name is shown, separated by '/'
+ * If the vdev has an associated path, which is typical of leaf vdevs,
+ * then the path is added.
+ * It would be nice to have the devid instead of the path, but under
+ * Linux we cannot be sure a devid will exist and we'd rather have
+ * something than nothing, so we'll use path instead.
+ */
+static char *get_vdev_desc(nvlist_t *nvroot, const char *parent_name)
```

Comment:
Unused function

Reply:
Don't even know why I left it there. Gone.